### PR TITLE
Fix issue with PeekingIterator where calling next() before peek() returns nil, in appropriately

### PR DESCRIPTION
**Pull request by cgrindel**
- Added test to demonstrate the issue.
- Updated next() to call peek to set temp

### CODE REVIEW

File: Tests/IteratorsTests/PeekingIteratorTests.swift [hunk 1]
```diff
@@ -13,4 +13,13 @@ final class PeekingIteratorTests: XCTestCase {
...
     }
+
+    func testPeekingIterator_WithoutPeek() {
+        // Make sure that iterator does not require peek to be called first.
+        var iterator = [1,2,3].makeIterator().peekable()
+        XCTAssertEqual(1, iterator.next())
+        XCTAssertEqual(2, iterator.next())
+        XCTAssertEqual(3, iterator.next())
+        XCTAssertEqual(nil, iterator.next())
+    }
```
Comment (atfelix) [inline Tests/IteratorsTests/PeekingIteratorTests.swift]: Let's add one more test that calls `next` first as well as `peek at some point.  Something like

```swift
func testCallingNextFirstAndPeekAtSomePoint() {
    var iterator = [1, 2, 3].makeIterator().peekable()
    XCTAssertEqual(1, iterator.next())
    XCTAssertEqual(2, iterator.peek())
    XCTAssertEqual(2, iterator.next())
    XCTAssertEqual(3, iterator.next())
    XCTAssertEqual(nil, iterator.next())
}
```

File: Tests/IteratorsTests/PeekingIteratorTests.swift [hunk 2]
```diff
@@ -13,4 +13,13 @@ final class PeekingIteratorTests: XCTestCase {
         XCTAssertEqual(nil, iterator.peek())
         XCTAssertEqual(nil, iterator.next())
     }
+
+    func testPeekingIterator_WithoutPeek() {
```
Comment (atfelix) [inline Tests/IteratorsTests/PeekingIteratorTests.swift]: nit:

```suggestion
    func testPeekingIteratorNeverCallingPeek() {
```

File: Tests/IteratorsTests/PeekingIteratorTests.swift [hunk 3]
```diff
@@ -13,4 +13,13 @@ final class PeekingIteratorTests: XCTestCase {
...
+        // Make sure that iterator does not require peek to be called first.
```
Comment (atfelix) [inline Tests/IteratorsTests/PeekingIteratorTests.swift]: I think the above test name suggestion takes care of this comment.

```suggestion
```

**Review by atfelix — COMMENTED** [review]
LGTM.  I have a few comment suggestions

File: Tests/IteratorsTests/PeekingIteratorTests.swift (hunk 1 above)
Comment (cgrindel) [reply Tests/IteratorsTests/PeekingIteratorTests.swift]: Done.

File: Tests/IteratorsTests/PeekingIteratorTests.swift (hunk 2 above)
Comment (cgrindel) [reply Tests/IteratorsTests/PeekingIteratorTests.swift]: Done.

File: Tests/IteratorsTests/PeekingIteratorTests.swift (hunk 3 above)
Comment (cgrindel) [reply Tests/IteratorsTests/PeekingIteratorTests.swift]: Done.

**Comment by cgrindel** [comment]
Hi, @atfelix , I updated the PR with your suggestions. Can you take another look?

**Review by atfelix — APPROVED** [review]

**Pull request merged**